packages/firestore/src/platform_browser/webchannel_connection.ts

```diff
@@ -123,12 +123,25 @@ export class WebChannelConnection implements Connection {
                 xhr.getResponseText()
               );
               if (status > 0) {
-                reject(
-                  new FirestoreError(
-                    mapCodeFromHttpStatus(status),
-                    'Server responded with status ' + xhr.getStatusText()
-                  )
-                );
+                const responseError = xhr.getResponseJson().error;
+                if (!!responseError) {
+                  const firestoreErrorCode = mapCodeFromHttpReponse(
```

Comment:
Reponse => Response

(Optional) I also wonder if we could/should factor out a bit more code.  Maybe call it `firestoreErrorFromRestError()` and pass in the full `responseError` value and return a full `FirestoreError` object?

As-is, this should technically be named something like mapCodeFromHttpResponseErrorStatusCode() or something...

Reply:
Doing so would require that we define the type for the responseError or pass it in as `any` with a ts-ignore line. I also added some checks to make sure that the fields are defined.